### PR TITLE
GTJ-26

### PR DESCRIPTION
**Pull request by gulyaasjaanos**
no tests yet
db-migrate added
service and repo code restructured

### CODE REVIEW

**Comment by gulyaasjaanos** [comment]
endpoint test added
simple function tests added

**Comment by gulyaasjaanos** [comment]
service-level test added

File: backend/package.json [hunk 1]
```diff
@@ -7,16 +7,19 @@
     "test": "jest --watchAll",
     "test:ci": "jest --ci --collectCoverage --reporters=default",
     "test:cover": "jest --collectCoverage",
-    "start": "nodemon --exec babel-node src/index.js",
+    "start": "nodemon --exec babel-node -r ./node_modules/dotenv/config src/index.js",
     "build": "babel src -d dist",
-    "serve": "node dist/index.js"
+    "serve": "node dist/index.js",
+    "db-migrate": "node node_modules/db-migrate/bin/db-migrate"
```
Comment (adamgyulavari) [inline backend/package.json:13]: @gulyaasjaanos 
```suggestion
    "db": "db-migrate"
```

File: backend/src/controllers/registerController.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { registerService } from '../services';
+
+export const registerController = (req,res) => {
+    registerService(req.body)
+     .then( response => res.json(response) )
+     .catch( error => res.json({error}) );
+};
```
Comment (adamgyulavari) [inline backend/src/controllers/registerController.js:7]: @gulyaasjaanos have new line at the end of file

File: backend/src/routes/api.routes.js [hunk 3]
```diff
@@ -10,4 +10,6 @@ router.use(bodyParser.json());
 
 router.get('/hello', helloController.get);
 
+router.post('/register', registerController);
```
Comment (adamgyulavari) [inline backend/src/routes/api.routes.js:13]: @gulyaasjaanos lets use `/users`

File: backend/src/data/connection.js [hunk 4]
```diff
@@ -20,8 +20,8 @@ export const db = {
 
           return;
         }
-
-        resolve({ results, fields });
+        const specialReturn = (special) ? results[special] : '';
+        resolve({ results, fields, specialReturn  });
```
Comment (adamgyulavari) [inline backend/src/data/connection.js:24]: @gulyaasjaanos lets just use the results instead

File: backend/src/index.js [hunk 5]
```diff
@@ -1,8 +1,7 @@
+require('dotenv').config();
```
Comment (adamgyulavari) [inline backend/src/index.js:1]: @gulyaasjaanos you can remove this

File: backend/src/repos/registerData.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import {db} from '../data/connection';
+
+const registerQuery = {
```
Comment (adamgyulavari) [inline backend/src/repos/registerData.js:3]: userQueries, kingdomQueries, insertQueries

File: backend/src/repos/registerData.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    user : 'INSERT INTO user (name,password) VALUES(?,?)',
+    kingdom : 'INSERT INTO kingdom (name) VALUES(?)',
+    user_kingdom : 'INSERT INTO user_kingdom (user_id,kingdom_id) VALUES(?,?)'
+};
+
+export const registerData = (table,params) => {
```
Comment (adamgyulavari) [inline backend/src/repos/registerData.js:9]: @gulyaasjaanos `save`

File: backend/src/services/registerService.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+export const registerService = (input) => {
+    return new Promise( async (resolve,reject) => {
+        const invalidInput = filterInput(input);
+        if ( invalidInput ) {
+            reject(invalidInput);
+        } else {
+            try {
+                const userid = await registerData('user',[input.username,input.password]);
```
Comment (adamgyulavari) [inline backend/src/services/registerService.js:18]: @gulyaasjaanos `db.save('user', {username: input.username, password: input.password})`
- db should check if the user to be saved is valid

File: backend/src/controllers/userController.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { userService } from '../services';
+
+export const userController = (req,res) => {
```
Comment (adamgyulavari) [inline backend/src/controllers/userController.js:3]: @gulyaasjaanos this should be an object with one property

File: backend/src/controllers/userController.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+    userService(req.body)
+     .then( response => res.json(response) )
```
Comment (adamgyulavari) [inline backend/src/controllers/userController.js:5]: @gulyaasjaanos success => 201 Created

File: backend/src/controllers/userController.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+     .catch( error => res.json({error}) );
```
Comment (adamgyulavari) [inline backend/src/controllers/userController.js:6]: @gulyaasjaanos fail => 400 Bad request

**Pull request merged**